docs/sql/statements/copy-from.rst

```diff
@@ -287,6 +289,38 @@ Using the ``s3://`` scheme automatically sets the `shared`_ to true.
    0.51.x these connections are using the HTTPS protocol. Please make sure you
    update your firewall rules to allow outgoing connections on port ``443``.
 
+.. _sql-copy-from-azblob:
+
+``azblob``
+''''''''''
+
+You can use the ``azblob:///`` scheme to access files on the `Azure Blob Storage`_.
+
+URI must consist of scheme and absolute path to the resource.
+
+For example:
+
+.. code-block:: text
+
+    azblob:///dir1/dir2/file1.json
+
+All other parameters must be provided in the ``WITH`` clause.
+
+For example:
+
+.. code-block:: text
+
+    COPY source
+    TO DIRECTORY 'azblob:///dir1/dir2'
+    WITH (
+        container = 'container',
```

Comment:
Why is container separated and not part of the URI? Doesn't seem quite consistent with how this is handled on Azure.

Reply:
Went in this direction (all in `WITH`) as we already separate 2 params and keeping all granular was not a big effort, and could be also more convenient for users. 

For example, `az storage container create` accepts endpoint AND account/key as a separate params (even if account is a part of endpoint). 

We can also support URI with account/container, smth like
 
 `https://myaccount.blob.core.windows.net/mycontainer/myvirtualdirectory/myblob`, plan is to discuss it later and maybe actually keep all-in-WITH and instead adjust S3.

Reply:
I think it is less convenient and also eg doesn't follow what other solutions accept.

Container is always required. Everything else in `WITH` could be optional.

Also if we adjust s3 import typically paths are provided as `s3://<bucket>/<blob>` (eg in the AWS Console)